EIPS/draft_token_validation.md

```diff
@@ -0,0 +1,181 @@
+## Preamble
+
+    EIP: <to be assigned>
+    Title: Token Validation
+    Author: Tom Carchrae<tom@finhaven.com>, Gleb Naumenko<gleb@finhaven.com>, Brooklyn Zelenka<brooklyn@finhaven.com>
+    Type: Standard Track
+    Category: ERC
+    Status: Draft
+    Created: 2018-02-14
+
+## Simple Summary
+A protocol for services providing financial transaction validation.
+
+## Abstract
+This standard provides a registry contract method for authorizing token transfers.
+By nature, this covers both initially issuing tokens to users (ie: transfer from contract to owner),
+transferring tokens between users, and token spends.
+
+## Motivation
+The tokenization of assets has wide application,
+not least of which is financial instruments such as securities.
+Most jurisdictions have placed legal constraints on what may be traded,
+and who can hold such tokens which are regarded as securities. Broadly this includes KYC and AML validation,
+but may also include time-based spend limits, total volume of transactions, and so on.
+
+Regulators and sanctioned third-party compliance agencies need some way to link
+off-chain compliance information such as identity and residency to an on-chain service.
+The application of this design is broader than legal regulation, encompassing all manner
+of business logic permissions for the creation, management, and trading of tokens.
+
+## Specification
+
+### TokenValidator
+
+```solidity
+interface TokenValidator {
+  function check(
+    address _token,
+    address _user
+  ) public returns(uint8 result)
+
+  function check(
+    address _token,
+    address _from,
+    address _to,
+    uint256 _amount
+  ) public returns (uint8 result)
+}
+```
+
+#### Methods
+
+##### check/2
+
+`function check(address token, address user) public returns (uint8 resultCode)`
+
+> parameters
+> * `token`: the token under review
+> * `user`: the user to check
+>
+> *returns* an 8-bit status code
+
+##### check/4
+
+`function check(address token, address from, address to, uint256 amount) public returns (uint8 resultCode)`
```

Comment:
Yeah, fair point. It might be the Elixir/Erlang in me (which has a very similar name overloading system). It reads as implicit to me, but my background is _very weird_ compared to most people 😛 `check(address)` and `check(to, from, amount)` seem pretty clear, but it also doesn't hurt to have some "duplication" (if it can even really be called that).


Thinking out loud, since all that the ABI knows about is the types, we really have:

* `check(address, address)`
* `check(address, address, address, uint8)`

Giving them longer/specific names would prevent namespace conflicts with like... 

* `check(contractA, contractB) { return contractA == contractB; }`
* `check(exchange, tokenA, tokenB, proposedConversionRate)`

🤔

Reply:
i like the elegance/terseness of `check` but i am not sure how it all plays with solidity.  

it seems that you're saved from namespace inheritance clashes  http://solidity.readthedocs.io/en/develop/contracts.html#overload-resolution-and-argument-matching
> If there is not exactly one candidate, resolution fails.

but joy!

> Return parameters are not taken into account for overload resolution.

Reply:
Well, we both like `check`. We could toss it out to the community and see what they think. I'm not married to `check(...)` 🤷🏻‍♀️